core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3791,6 +3791,12 @@ private static AtomicLong getAtomicLong(String key) {
     return atomic;
   }
 
+  /** Support the ARRAY_CONTAINS function. */
+  public static boolean contains(List list, Object element) {
+    final Set set = new HashSet(list);
```

Comment:
Is this for faster search?

Reply:
yeap, it is no different with for loop, may cost some space. if you think it is need, i will change it to for loop

Reply:
In fact, set has not been reused, and I don't think it can improve speed.

Because the construction time of the set may be longer than the search time of the list.

Reply:
yeap, fixed @JiajunBernoulli

Reply:
I don't think you need this method at all. The code generator can just call `java.util.List.contains`. You will need to add it to `BuiltInMethod`.

Reply:
@julianhyde I have considered this， but it can't. because the array_contains have 2 args instead of 1 in java.util.List.contains